libs/labelbox/src/labelbox/schema/member.py

```diff
@@ -0,0 +1,433 @@
+            "userGroups": [],
+            "projectMemberships": [],
+            "defaultRole": None,
+        }
+        model = self.model_copy(update=Member(**user).model_dump())
+
+        for userGroup in result["user"]["userGroups"]["nodes"]:
+            model.user_group_ids.add(userGroup["id"])
+
+        model.default_role = Role(
+            self.client,
+            field_values={
+                "id": result["user"]["defaultRole"]["id"],
+                "name": result["user"]["defaultRole"]["name"],
+            },
+        )
+
+        for project_membership in result["projectMemberships"]:
+            project_membership["role"] = Role(
+                self.client,
+                field_values={
+                    "id": project_membership["role"]["id"],
+                    "name": project_membership["role"]["name"],
+                },
+            )
+            project_membership["projectId"] = project_membership["project"][
+                "id"
+            ]
+            model.project_memberships.add(
+                ProjectMembership(**project_membership)
+            )
+
+        return model
+
+    def update(self) -> "Member":
+        """
+        Updates the member in Labelbox.
+
+        Returns:
+            Member: The updated Member object. (self)
+
+        Raises:
+            ResourceNotFoundError: If the update fails due to unknown member
+            UnprocessableEntityError: If the update fails due to a malformed input
+            ValueError: If the member id is current member id.
+        """
+        query = """
+            mutation SetUserAccessPyApi($id: ID!, $roleId: ID!, $canAccessAllProjects: Boolean!, $groupIds: [String!], $projectMemberships: [ProjectMembershipsInput!]) {
+            setUserAccess(
+                where: {id: $id}
+                data: {roleId: $roleId, canAccessAllProjects: $canAccessAllProjects, groupIds: $groupIds, projectMemberships: $projectMemberships}
+            ) {
+                id
+                __typename
+            }
+            }
+        """
+        params = {
+            "id": self.id,
+            "roleId": self.default_role.uid if self.default_role else None,
+            "canAccessAllProjects": self.can_access_all_projects,
+            "groupIds": list(self.user_group_ids),
+            "projectMemberships": [
+                project_membership.model_dump()
+                for project_membership in self.project_memberships
+            ],
+        }
+
+        if self.id == self._current_user_id:
+            raise ValueError("Unable to update self")
+
+        try:
+            result = self.client.execute(query, params, experimental=True)
+            if not result:
+                raise ResourceNotFoundError(
+                    message="Failed to update member as member does not exist"
+                )
+        except MalformedQueryException as e:
+            raise UnprocessableEntityError("Failed to update member") from e
```

Comment:
this is already handled by `.execute` and raising new exceptions would just hide original error messages

Reply:
Fixed